public/app/core/components/UserEdit/Admin/UserOrganizations.tsx

```diff
@@ -0,0 +1,194 @@
+import React, { PureComponent } from 'react';
+import _ from 'lodash';
+import { hot } from 'react-hot-loader';
+import { connect } from 'react-redux';
+import { getBackendSrv } from '@grafana/runtime';
+import { StoreState, OrgUser } from 'app/types';
+import { FormLabel, Select, Button } from '@grafana/ui';
+import { OrganizationPicker, OrgOption } from 'app/core/components/Select/OrganizationPicker';
+
+export interface RoleOption {
+  value: string;
+  label: string;
+}
+
+export interface UserOrg {
+  orgId: number;
+  name: string;
+  role: string;
+}
+
+export interface Props {
+  userId: number;
+  user: OrgUser;
+}
+
+export interface State {
+  newOrg: OrgOption;
+  newOrgRole: string;
+  roles: RoleOption[];
+  userOrgs: UserOrg[];
+}
+
+export class UserOrganizations extends PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      newOrg: {} as OrgOption,
+      newOrgRole: 'Editor',
+      roles: [
+        {
+          label: 'Viewer',
+          value: 'Viewer',
+        },
+        {
+          label: 'Editor',
+          value: 'Editor',
+        },
+        {
+          label: 'Admin',
+          value: 'Admin',
+        },
+      ],
+      userOrgs: [],
+    };
+  }
+
+  async componentDidMount() {
+    await this.loadUserOrgs();
+  }
+
+  onRoleSelect = (newOrgRole: string) => {
+    this.setState({ newOrgRole });
+  };
+
+  onOrgSelect = (org: OrgOption) => {
+    this.setState({ newOrg: org });
+  };
+
+  async onOrgRoleChange(userOrg: UserOrg, newRole: string) {
+    const { userId } = this.props;
+    userOrg.role = newRole;
+    await getBackendSrv().patch('/api/orgs/' + userOrg.orgId + '/users/' + userId, userOrg);
+    await this.loadUserOrgs();
```

Comment:
This piece of code is repeated in all of the methods here (onOrgRoleChange, onAddToOrg, removeFromOrg).  I would like the actual backend cals to be passed down from a component responsible for backend interaction only - meaning that the users re-fetching (loadUsersOrg) would be transparent to this component and encapsulated in some kind of user organisations API provider . This is sth that I think we struggle a bit not only here, as we load the data from within presentational components. Any thoughts on that @torkelo ?